x/ccv/provider/keeper/relay_test.go

```diff
@@ -715,3 +715,65 @@ func TestSendVSCPacketsToChainFailure(t *testing.T) {
 	// Pending VSC packets should be deleted in StopConsumerChain
 	require.Empty(t, providerKeeper.GetPendingVSCPackets(ctx, "consumerChainID"))
 }
+
+// TestOnTimeoutPacketWithNoChainFound tests the `OnTimeoutPacket` method fails when no chain is found
+func TestOnTimeoutPacketWithNoChainFound(t *testing.T) {
+	// Keeper setup
+	providerKeeper, ctx, ctrl, _ := testkeeper.GetProviderKeeperAndCtx(t, testkeeper.NewInMemKeeperParams(t))
+	defer ctrl.Finish()
+
+	// We do not `SetChannelToChain` for "channelID" and therefore `OnTimeoutPacket` fails with no chain found
+	packet := channeltypes.Packet{
+		SourceChannel: "channelID",
+	}
+	err := providerKeeper.OnTimeoutPacket(ctx, packet)
+	require.Error(t, err)
+}
+
+// TestOnTimeoutPacketStopsChain tests that the chain is stopped in case of a timeout
+func TestOnTimeoutPacketStopsChain(t *testing.T) {
+	// Keeper setup
+	providerKeeper, ctx, ctrl, mocks := testkeeper.GetProviderKeeperAndCtx(t, testkeeper.NewInMemKeeperParams(t))
+	defer ctrl.Finish()
+	providerKeeper.SetParams(ctx, providertypes.DefaultParams())
+
+	testkeeper.SetupForStoppingConsumerChain(t, ctx, &providerKeeper, mocks, false)
```

Comment:
Since we're calling `SetupForStoppingConsumerChain` s.t. no mocks for `StopConsumerChain` are expected, this test is only asserting that `OnTimeoutPacket` doesn't return an error.

Ie. I could comment out the `StopConsumerChain` call in `OnTimeoutPacket` and this test would still pass. 

Therefore the comment is misleading

> tests that the chain is stopped in case of a timeout

Reply:
If the consumer chain is stopped, then we expect that the provider's state is cleaned (see [spec methods](https://github.com/cosmos/ibc/blob/jk/ccvTLA/spec/app/ics-028-cross-chain-validation/methods.md#consumer-chain-removal)), so using `TestProviderStateIsCleanedAfterConsumerChainIsStopped` to test that the state is cleaned.